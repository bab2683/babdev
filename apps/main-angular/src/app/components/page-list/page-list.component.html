<ul class="page-list">
  <li class="page-list__item" *ngFor="let page of pages">
    <a class="page-list__link link-effect-1" [routerLink]="page.url">{{
      page.key | translate | async
    }}</a>
  </li>
</ul>
