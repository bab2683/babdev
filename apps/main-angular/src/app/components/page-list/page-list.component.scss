@import "~@styleguide/_functions";

.page-list {
  &__item {
    & + & {
      margin-top: px-to-rem(10);
    }
  }
}
