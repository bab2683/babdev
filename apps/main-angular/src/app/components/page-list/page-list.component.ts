import { Component } from '@angular/core';

import { Pages } from '@constants';

@Component({
  selector: 'babdev-page-list',
  templateUrl: './page-list.component.html',
  styleUrls: ['./page-list.component.scss']
})
export class PageListComponent {
  pages = Pages;
}
