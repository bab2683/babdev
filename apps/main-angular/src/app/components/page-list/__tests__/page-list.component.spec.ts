import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterLinkDirectiveStub } from '@babdev/testing';
import { TranslatePipeMock } from '@babdev/translate-testing';

import { PageListComponent } from '../page-list.component';

describe('PageListComponent', () => {
  let component: PageListComponent;
  let fixture: ComponentFixture<PageListComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [PageListComponent, RouterLinkDirectiveStub, TranslatePipeMock]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(PageListComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
