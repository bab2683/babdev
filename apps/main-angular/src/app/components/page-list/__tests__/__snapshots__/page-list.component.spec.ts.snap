// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageListComponent should match snapshot 1`] = `
<babdev-page-list
  pages={[Function Array]}
>
  <section
    class="page-list"
  >
    <a
      class="page-list__home"
      routerlink="/"
    >
      Braulio
    </a>
    <ul
      class="page-list__list"
    >
      
      <li
        class="page-list__item"
      >
        <a
          class="page-list__link link-effect-1"
        >
          common.PAGES.CONTACTS
        </a>
      </li>
      <li
        class="page-list__item"
      >
        <a
          class="page-list__link link-effect-1"
        >
          common.PAGES.EXPERIENCE
        </a>
      </li>
    </ul>
  </section>
</babdev-page-list>
`;
